### PR TITLE
sink: implement kafka sink

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
implement kafka sink

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: cdc/sink/black_hole.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	log.Info("BlockHoleSink: Checkpoint Event", zap.Uint64("checkpoint ts", ts))
+	return nil
+}
+
+func (b *blackHoleSink) EmitRowChangedEvent(ctx context.Context, rows ...*model.RowChangedEvent) error {
+	for _, row := range rows {
+		if row.Resolved {
+			atomic.StoreUint64(&b.checkpointTs, row.Ts)
+		}
+		log.Info("BlockHoleSink: Row Changed Event", zap.Any("row", row))
```
Comment (amyangfei) [inline cdc/sink/black_hole.go]:
```suggestion
```
There will be too many logs, suggest to make it more like a blackhole.

File: cdc/sink/mq.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+func (k *mqSink) Run(ctx context.Context) error {
+	for {
+		var sinkCheckpointTs uint64
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case sinkCheckpointTs = <-k.sinkCheckpointTsCh:
+		}
+		globalResolvedTs := atomic.LoadUint64(&k.globalResolvedTs)
+		if globalResolvedTs < sinkCheckpointTs {
```
Comment (amyangfei) [inline cdc/sink/mq.go]: In which case will `globalResolvedTs` be less than `sinkCheckpointTs`

File: cdc/sink/mysql.go [hunk 3]
```diff
@@ -80,12 +80,40 @@ func (s *mysqlSink) EmitRowChangedEvent(ctx context.Context, rows ...*model.RowC
 		//	continue
 		//}
 		key := util.QuoteSchema(row.Schema, row.Table)
+		checkDecode(row)
```
Comment (amyangfei) [inline cdc/sink/mysql.go]: Is this debug code?

File: cdc/sink/sink.go [hunk 4]
```diff
@@ -15,65 +15,53 @@ package sink
...
+	}
+	switch strings.ToLower(sinkURI.Scheme) {
+	case "blackhole":
+		return newBlackHoleSink(), nil
+	case "mysql", "tidb":
+		return newMySQLSink(sinkURI, nil, opts)
+	case "kafka":
+		return newKafkaSaramaSink(sinkURI)
+	default:
+		return nil, errors.New(fmt.Sprintf("the sink scheme (%s) is not supported", sinkURI.Scheme))
```
Comment (amyangfei) [inline cdc/sink/sink.go]:
```suggestion
		return nil, errors.Errorf("the sink scheme (%s) is not supported", sinkURI.Scheme)
```

File: cdc/sink/mysql.go (hunk 3 above)
Comment (leoppro) [reply cdc/sink/mysql.go]: yes, i will remove it

File: cdc/sink/mq.go (hunk 2 above)
Comment (leoppro) [reply cdc/sink/mq.go]: https://github.com/pingcap/ticdc/pull/322/files#diff-172dd021d09a9fac1ffc69e5792d4c5bR83

sinkCheckpointTs is localResolvedTs in the processor, when we add a table, old local resolvedTS(min resolvedTs of old tables) may be greater than resolvedTS of the new table. At this moment, local resolvedTS will be fallback.

when local resolvedTS is fallback, we will postpone to pushing global resolvedts:
https://github.com/pingcap/ticdc/blob/master/cdc/processor.go#L526

**Comment by leoppro** [comment]
PTAL again @amyangfei

File: cdc/sink/mq.go (hunk 2 above)
Comment (amyangfei) [reply cdc/sink/mq.go]: Oh I see. Also adding some comments here.

Comment (leoppro) [reply cdc/sink/mq.go]: added

**Pull request merged**